resources/web/wwi/Stream.js

```diff
@@ -57,7 +57,27 @@ export default class Stream {
 
   _onSocketMessage(event) {
     let data = event.data;
-    if (data.startsWith('robot:') || data.startsWith('robot window:'))
+
+    if (data.startsWith('robot window:')) {
+      const json = JSON.parse(data.substring(14));
+      const robotWindow = json.window === '<generic>' ? 'generic' : json.window;
+
+      if (json.remove) {
+        const robots = webots.currentView.robots;
+        let index;
+        for (let i = 0; i < robots.length; i++) {
+          if (robots[i].name === json.robot) {
+            index = i;
+            break;
+          }
+        }
+        if (typeof index !== 'undefined')
+          robots.splice(index, 1);
```

Comment:
Isn't that simpler?
```suggestion
        for (let i = 0; i < robots.length; i++) {
          if (robots[i].name === json.robot) {
            robots.splice(i, 1);
            break;
          }
        }
```

Reply:
I will test it. 
I don't usually like modifying the object I am iterating on, but as we break immediately, it should be ok